### PR TITLE
Fixes #9704,#9705 - ability to customize vmware VM on image-based provisioning

### PR DESCRIPTION
**Pull request by iNecas**
Allows applying the changes form the form to the VM when cloning from a
template.

Depends on https://github.com/fog/fog/pull/3570

### CODE REVIEW

**Comment by domcleal** [comment]
Setting to WoC until a version of Fog with the patch is available.  If merged, it may be blocked on http://projects.theforeman.org/issues/10470.

**Comment by iNecas** [comment]
The fog patch got merged, I've rebased. We're probably still waiting for new fog release first

File: app/models/compute_resources/foreman/model/vmware.rb [hunk 1]
```diff
@@ -351,14 +351,10 @@ def new_vm(args = {})
     # Fog calls +cluster.resourcePool.find("Resources")+ that actually calls
     # +searchIndex.FindChild("Resources")+ in RbVmomi that then returns nil
     # because it has no children.
-    def clone_vm(args)
-      args = parse_args args
+    def clone_vm raw_args
```
Comment (shlomizadok) [inline app/models/compute_resources/foreman/model/vmware.rb]: nitpick: parentheses

**Comment by dLobatog** [comment]
@iNecas Fog release is out, will probably be merged soon (https://github.com/theforeman/foreman/pull/2602)

**Comment by iNecas** [comment]
Rebased and addressed the comments

**Comment by dLobatog** [comment]
@shlomizadok Mind to test this one?

**Comment by shlomizadok** [comment]
Sure :)

2015-08-25 17:17 GMT+03:00 Daniel Lobato García notifications@github.com:

**Comment by shlomizadok** [comment]
Works as expected

**Comment by iNecas** [comment]
Thanks @shlomizadokl, Amy and Aris :)

**Comment by dLobatog** [comment]
Merged as e4fb003a25a6065320f0412a1de944e181d40ad1, thanks @iNecas!

**Comment by shlomizadok** [comment]
[IMAGE ALT TEXT HERE](http://www.youtube.com/watch?v=nMO5Ko_77Hk)
@iNecas - this is for you :)

**Pull request closed without merging**